files/lib/data/conversation/ConversationAction.class.php

```diff
@@ -216,6 +216,10 @@ public function markAsRead() {
 		if (empty($this->parameters['visitTime'])) {
 			$this->parameters['visitTime'] = TIME_NOW;
 		}
+
+		if (!isset($this->parameters['userID'])) {
+			$this->parameters['userID'] = WCF::getUser()->userID;
```

Comment:
This introduces a security vulnerability. A user may maliciously pass in the userID of another user of any conversation they are a member of.